src/consensus/aft/raft.h

```diff
@@ -1534,34 +1537,49 @@ namespace aft
       // commit to that idx.
       auto new_commit_idx = std::numeric_limits<Index>::max();
 
-      for (auto& c : configurations)
+      // Check if we are using BFT consensus and we have at any point at least 3
+      // nodes in the network. The requirement for 3 nodes ensures that we can
+      // start a new network which in the beginning will only start with 1 node.
+      if (
+        consensus_type == ConsensusType::BFT &&
+        (node_count() >= min_bft_node_count ||
+         state->should_start_using_byz_commit))
       {
-        // The majority must be checked separately for each active
-        // configuration.
-        std::vector<Index> match;
-        match.reserve(c.nodes.size() + 1);
-
-        for (auto node : c.nodes)
+        auto progress_tracker = store->get_progress_tracker();
+        CCF_ASSERT(progress_tracker != nullptr, "progress_tracker is not set");
+        new_commit_idx = progress_tracker->get_highest_commit_level();
+        state->should_start_using_byz_commit = true;
```

Comment:
Something I am missing here: at the very beginning, `node_count() < min_bft_node_count` so we only get here if `state->should_start_using_byz_commit == true`. However, it seems that `state->should_start_using_byz_commit` is initially `false` and only becomes `true` if we get here, which isn't possible because `node_count() < min_bft_node_count`?

Reply:
As discussed, this is actually working the other way around: the commit logic defaults to `CFT` until we have `min_bft_node_count` nodes and then uses `BFT` even if `node_count()` goes under `min_bft_node_count`. We should probably expand the comment to reflect that.